bids/layout/layout.py

```diff
@@ -443,6 +463,24 @@ def parse_file_entities(self, filename, scope='all', entities=None,
         return parse_file_entities(filename, entities, config,
                                    include_unmatched)
 
+    def _get_derivative_dirs(self, paths):
+        def check_for_description(path):
+            return (path/"dataset_description.json").exists()
```

Comment:
Personal preference to use `__class__.method(expr)` over `(expr).method()`.

```suggestion
            return Path.exists(path / "dataset_description.json")
```